lib/datadog/tracing/contrib/karafka/patcher.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative '../patcher'
+require_relative 'events'
+require_relative 'ext'
+require_relative 'distributed/propagation'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Karafka
+        module MessagesPatch
+          def configuration
+            Datadog.configuration.tracing[:sidekiq]
+          end
+
+          def propagation
+            @propagation ||= Contrib::Sidekiq::Distributed::Propagation.new
+          end
+
+          def each(&block)
+            @messages_array.each do |message|
```

Comment:
Regarding what you wrote here @mensfeld https://karafka.io/docs/Consuming-Messages/#in-batches, this messages enumerator is supposed to be the only API interface that our users use for batch operations, am I right? Seems I can't see any alternative ways in this doc, except you're saying these high-scale users delegate the whole messages to another process by saving them to a storage (e.g., like Event.insert_all messages) without calling .each ?

Reply:
> this messages enumerator is supposed to be the only API interface that our users use for batch operations, am I right?

No.

> except you're saying these high-scale users delegate the whole messages to another process by saving them to a storage (e.g., like Event.insert_all messages) without calling .each ?

Yes. Alongside you can use `Messages#payloads` directly omitting the headers when not needed. `Messages#each` is a popular use-case but not the only one. While Karafka provides primitives for both, it my itself makes no assumptions about that nature of the processing, that's why `#each` is not instrumented on my side. The moment I `yield` control to the user, it is on the user to define the nature of the processing.

Reply:
Thanks @mensfeld, I see your point, but honestly, this seems like the most suitable place for me to patch and enable distributed tracing, if the users (who also use Datadog like me) want to use #payloads to do their batching implementation, then loosing distributed tracing, in that case, would be reasonable. 

I understand that the `#each` method in the Karafka gem isn’t instrumented since neither you nor the gem itself needs to expose it. However, this distributed tracing is specific to the `datadog` gem and aligns perfectly with my motivation (in my PR description). So would love to hear thoughts on this from the Datadog team as well. @marcotc 

willing to address if there is another place that we should patch to have our distributed tracing instead.

Reply:
> willing to address if there is another place that we should patch to have our distributed tracing instead.

The only thing here is a philosophy of tracing. Ideally if we could have both and allow users to go with the approach they are taking for processing their data, then it would be ideal.

FYI I'm not sure if I mentioned this but feel free to ping me if you need any more help. Also happy to help with this work via Slack/etc if needed.